pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -740,10 +749,10 @@ func (o *MustGatherOptions) newPod(node, image string, hasMaster bool) *corev1.P
 		},
 		Spec: corev1.PodSpec{
 			NodeName: node,
-			// This pod is ok to be OOMKilled but not preempted. Following the conventions mentioned at:
+			// This pod is ok to be OOMKilled and preempted. Following the conventions mentioned at:
 			// https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md#priority-classes
-			// so setting priority class to system-cluster-critical
-			PriorityClassName: "system-cluster-critical",
+			// so setting priority class to openshift-user-critical
+			PriorityClassName: "openshift-user-critical",
```

Comment:
@ingvagabund would have any reservations about changing this to `openshift-user-critical`?

Reply:
My only reservation would be inability to deploy a must-gather pod in cases the monitoring workload eats up all the available resources. For which one will not be able to run the must-gather. Nevertheless, that's a case that can be analyzed by collecting running `oc inspect` and `oc logs/get/...` commands.

Given all control plane components are of the highest priority and each master node has the minimum resources required, the scheduler will always be able to schedule new control plane components by preemption. The installer pods have [system-node-critical](https://github.com/openshift/library-go/blob/master/pkg/operator/staticpod/controller/installer/manifests/installer-pod.yaml#L47) pc.

Though, this will deserve few lines in the release notes so users are aware of this change.

Reply:
@mfojtik @soltysh any objections here?

Reply:
We're talking about it on slack, I'll update once we settle. For now I'm strongly against such a change.

Reply:
I have dug the kubernetes code base and this https://github.com/kubernetes/kubernetes/blob/02960a8253ecf89d0487bb8aacdf499c80331064/pkg/kubelet/eviction/eviction_manager.go#L573 prevents the `system-cluster-critical` priority classes are being evicted.
As a workaround to not change the priority class of must-gather;
* in a separete goroutine we can listen this log https://github.com/kubernetes/kubernetes/blob/02960a8253ecf89d0487bb8aacdf499c80331064/pkg/kubelet/eviction/eviction_manager.go#L574 via `oc adm node-logs` (assuming that this error message won't be changed in the future)
* When we see the error message, we manually kill the gather container via `oc exec -- kill 1` (there is no differentiator about the eviction reason and we assume that this is due to volume is exceeded)

This way would work as if user passes `--keep` because we are trying to keep the pod to let user downloads the logs.

@soltysh what do you think about this?

Reply:
But there is no guarantee that [this log](https://github.com/kubernetes/kubernetes/blob/02960a8253ecf89d0487bb8aacdf499c80331064/pkg/kubelet/eviction/eviction_manager.go#L574) will not be changed(or even removed).

I start thinking that what about we lower the priority class name to openshift-user-critical and for catastrophic cases, introduce a hidden flag(to use `system-cluster-critical` priority class name) by warning user about the possible risks.

Reply:
No matter which solution gets implemented, would it make sense to prioritize which resources get collected first? If the size gets limited, we should collect the most important resources first.

Reply:
> No matter which solution gets implemented, would it make sense to prioritize which resources get collected first? If the size gets limited, we should collect the most important resources first. So whatever gets left is still usable for analysis. To at least get some pointers where to look. E.g. collect pod logs first, then their manifests, then other resources?

Sounds like a good enhancement for the future releases. We need to reevaluate the greedy resource traversal and may introduce a priority base search perhaps.

Reply:
@soltysh as we discussed on Slack. I changed this PR in the way that as you suggested. Please let me know what you think, when you have a chance. Thanks.